File: src/main/java/com/tomatokey/framework/exception/ValidError.java
package com.tomatokey.framework.exception;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import org.springframework.http.HttpStatus;
import org.springframework.validation.FieldError;
import org.springframework.validation.ObjectError;

import java.io.Serializable;

@Getter
@Builder
@AllArgsConstructor
public class ValidError implements Serializable {

    private String field;
    private String message;

    public static ValidError of(FieldError error) {
        return ValidError.builder()
                .field(error.getField())
                .message(error.getDefaultMessage())
                .build();
    }

}

File: src/main/kotlin/com/prototype/framework/exception/ValidError.kt
package com.prototype.framework.exception

import org.springframework.validation.FieldError
import java.io.Serializable

data class ValidError(
    val field: String,
    val message: String?
) : Serializable {

    companion object {
        fun of(error: FieldError): ValidError {
            return ValidError(error.field, error.defaultMessage)
        }
    }

}